### PR TITLE
[4.0] Admin menu empty target

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue #20975.

### Summary of Changes

This prevents `target` attribute with empty value being added to links in administrator menu module.

### Testing Instructions

Inspect HTML markup generated by administrator menu module.

### Expected result

Empty `target` attribute not added, e.g. `<li role="menuitem"><a class="no-dropdown"  href="index.php"><span class="sidebar-item-title">Control Panel</span></a></li>`

### Actual result

Empty `target` attribute added, e.g. `<li role="menuitem"><a class="no-dropdown"  href="index.php" target=""><span class="sidebar-item-title">Control Panel</span></a></li>`

### Documentation Changes Required
No.

### CODE REVIEW

File: administrator/modules/mod_menu/tmpl/default_submenu.php [hunk 1]
```diff
@@ -82,13 +82,13 @@
 	$link = '#collapse' . $this->tree->getCounter();
 }
 
-if ($link !== null && $current->get('target') !== null)
+if ($link !== null && $current->get('target') !== null && $current->get('target') !== '')
```
Comment (Quy) [inline administrator/modules/mod_menu/tmpl/default_submenu.php:85]: How about `!empty($current->get('target'))`?

Comment (SharkyKZ) [reply administrator/modules/mod_menu/tmpl/default_submenu.php:85]: A matter of preference and whether `0` should pass. HTML validator allows `0`.

Comment (Quy) [reply administrator/modules/mod_menu/tmpl/default_submenu.php:85]: Will the value ever be 0?

Comment (SharkyKZ) [reply administrator/modules/mod_menu/tmpl/default_submenu.php:85]: We don't expose this in backend but `target` value can be a name of an `iframe` and `0` is a valid name.

Comment (Quy) [reply administrator/modules/mod_menu/tmpl/default_submenu.php:85]: https://html.com/attributes/a-target/
According to the above link, it refers to frame name and not iframe. Frames are deprecated in HTML5.

**Comment by ggppdk** [comment]
I have tested this item :white_check_mark: successfully on 29a2686cdf23d08edee699133784c42191d029b4<br /><br />Works, no more empty target="" (no more invalid HTML)

Also about target="0" , 0 is an valid frame name , the fact that frames are deprecated in HTML5, is not reason to handle target="0" differently than target="1"<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20976">issues.joomla.org/tracker/joomla-cms/20976</a>.</sub>

File: administrator/modules/mod_menu/tmpl/default_submenu.php (hunk 1 above)
Comment (SharkyKZ) [reply administrator/modules/mod_menu/tmpl/default_submenu.php:85]: This applies to iframes too. You can test this snippet. Clicking the link should open it in the iframe.
```
<!doctype html>
	<html>
		<head>
			<title>Test</title>
		</head>
		<body>
			<a href="https://html.com/attributes/a-target/" target="0">Click</a>
			<iframe name="0"></iframe>
		</body>
	</html>
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on be5cf43313420da0910a6d9ccac163d6bedc631d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20976">issues.joomla.org/tracker/joomla-cms/20976</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20976">issues.joomla.org/tracker/joomla-cms/20976</a>.</sub>

**Pull request merged**